app/src/main/java/com/example/awesomelanguagelearning/core/ui/views/TextWithClickablePart.kt

```diff
@@ -0,0 +1,95 @@
+package com.example.awesomelanguagelearning.core.ui.views
+
+import androidx.compose.foundation.text.ClickableText
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.tooling.preview.Preview
+import com.example.awesomelanguagelearning.core.ui.theme.AppTheme
+
+@Composable
+fun TextWithClickablePart(
+    regularText: String,
+    clickableText: String,
+    modifier: Modifier = Modifier,
+    textColor: Color = AppTheme.colors.grayMedium,
+    textStyle: TextStyle = AppTheme.typography.bodyM,
+    clickableTextColor: Color = AppTheme.colors.black,
+    clickableTextStyle: TextStyle = AppTheme.typography.bodyBoldM,
```

Comment:
@igel911 I think you can reduce the number of parameters and create something like 
```
class TextColors(val regular: Color, val clickable: Color)
class TextStyles(val regular: TextStyle, val clickable: TextStyle)
```

Reply:
@slothdeboss done